### PR TITLE
PLU-333: [PAGINATE-1] tiles pagination

### PR DESCRIPTION
**Pull request by pregnantboy**
## Changes
- Introduce filtering and pagination to the `getTables` GraphQL query in schema file.
- Modify query resolver to filter and paginate
- Add pagination component on frontend

### CODE REVIEW

**Comment by pregnantboy** [comment]
* **#758** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/758?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#773](https://github.com/opengovsg/plumber/pull/773) <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/773?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#774](https://github.com/opengovsg/plumber/pull/774) <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/774?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#757** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/757?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#756** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/756?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pregnantboy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/src/graphql/schema.graphql [hunk 1]
```diff
@@ -14,33 +14,33 @@ type Query {
...
     status: String
     searchInput: String
-  ): ExecutionConnection
+  ): PaginatedExecutions
   getExecutionSteps(
     executionId: String!
     limit: Int!
     offset: Int!
-  ): ExecutionStepConnection
+  ): PagingatedExecutionSteps
```
Comment (m0nggh) [inline packages/backend/src/graphql/schema.graphql]: typo boss
```suggestion
  ): PaginatedExecutionSteps
```

File: packages/backend/src/helpers/pagination.ts [hunk 2]
```diff
@@ -22,6 +22,7 @@ const paginate = async <TModel extends Model>(
       totalCount: count,
     },
     edges: records.map((record: TModel) => {
+      // TODO: remove this node key and return the record directly
```
Comment (m0nggh) [inline packages/backend/src/helpers/pagination.ts:25]: is this done in another PR?

File: packages/frontend/src/pages/Tiles/components/TileList.tsx [hunk 3]
```diff
@@ -33,11 +31,12 @@ import {
 } from '@opengovsg/design-system-react'
 
 import * as URLS from '@/config/urls'
+import { TableMetadata } from '@/graphql/__generated__/graphql'
```
Comment (m0nggh) [inline packages/frontend/src/pages/Tiles/components/TileList.tsx]:
```suggestion
import type { TableMetadata } from '@/graphql/__generated__/graphql'
```

File: packages/backend/src/graphql/queries/tiles/get-tables.ts [hunk 4]
```diff
@@ -1,15 +1,22 @@
...
   context,
 ) => {
-  const tables = await context.currentUser
+  const tables = context.currentUser
     .$relatedQuery('tables')
+    .where((builder) => {
+      if (name) {
+        builder.where('table_metadata.name', 'ilike', `%${name}%`)
+      }
+    })
```
Comment (m0nggh) [inline packages/backend/src/graphql/queries/tiles/get-tables.ts:16]: is this used in another PR? ok nvm it is in the next pr mb

**Review by m0nggh — APPROVED** [review]
tested that it works with more than 10 tiles and empty tile state

PS: actually i realised a bit scope creep but right now users can type `?page=4` and if users have fewer than 30 pipes, then maybe we should show a separate page instead? like just a simple check for `page > 1 && isEmpty` condition would suffice i thinkkk

File: packages/frontend/src/pages/Tiles/components/TileList.tsx (hunk 3 above)
Comment (pregnantboy) [reply packages/frontend/src/pages/Tiles/components/TileList.tsx]: TODO: add this to eslint rule

**Comment by pregnantboy** [comment]
### Merge activity

* **Oct 14, 6:27 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/756).
* **Oct 14, 6:27 AM EDT**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/756).
